### PR TITLE
fix(icgm): make sensor compatible and more aligned with simulator

### PR DESCRIPTION
**Pull request by scaubrey**
Added in 1) a few necessary compatibility changes for the simulator 2) some design changes 3) some stylistic changes. Would be great to review async or together ASAP.

### CODE REVIEW

File: tests/test_icgm_sensor_generator_refactor.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def test_refactor_default_state():
 
     icgm_sensor_generator = iCGMSensorGenerator(batch_training_size=3, true_dataset_name="48hours-sinusoid")
     icgm_sensor_generator.fit(true_bg_trace=test_bg_trace)
-    sensors = icgm_sensor_generator.generate_sensors(3)
+    sensors = icgm_sensor_generator.generate_sensors(sensor_start_datetime=datetime.datetime.now(), n_sensors=3)
```
Comment (scaubrey) [inline tests/test_icgm_sensor_generator_refactor.py]:
```suggestion
    sensors = icgm_sensor_generator.generate_sensors(sensor_start_datetime= TEST_DATETIME, n_sensors=3)
```

File: tests/test_icgm_sensor_generator_refactor.py [hunk 2]
```diff
@@ -57,7 +58,7 @@ def test_generator_fails_without_fit():
     icgm_sensor_generator = iCGMSensorGenerator()
 
     with pytest.raises(Exception) as e:
-        sensors = icgm_sensor_generator.generate_sensors(n_sensors=3)
+        sensors = icgm_sensor_generator.generate_sensors(n_sensors=3, sensor_start_datetime=datetime.datetime.now())
```
Comment (scaubrey) [inline tests/test_icgm_sensor_generator_refactor.py]:
```suggestion
        sensors = icgm_sensor_generator.generate_sensors(n_sensors=3, sensor_start_datetime=TEST_DATETIME)
```

**Review by jameno — APPROVED** [review]
LGTM!

**Comment by scaubrey** [comment]
Keeping `cas/icgm` branch open for now since the simulator refers to it for the Insulet pulse analysis

**Pull request merged**